SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -45,6 +55,128 @@ public MainWindow()
+                SYLMap.RemoveLastMarker();
+                if (updateMarketplacesWasClicked)
+                    SetMarkers(sender, e);
+            }
+            else
+            {
+                ErrorLabel1.Visibility = Visibility.Visible;
+            }
+        }
+
+        private void SaveData()
+        {
+            // Checking person's ID
+            List<double> personID = new List<double>();
+            using (TextFieldParser csvParser = new TextFieldParser(filePath))
+            {
+                csvParser.CommentTokens = new string[] { "#" };
+                csvParser.SetDelimiters(new string[] { "," });
+                csvParser.HasFieldsEnclosedInQuotes = true;
+
+                // Skip the row with the column names
+                csvParser.ReadLine();
+
+                while (!csvParser.EndOfData)
+                {
+                    // Read current line fields, pointer moves to the next line.
+                    string[] fields = csvParser.ReadFields();
+                    personID.Add(Double.Parse(fields[4]));
+                }
+            }
+
+            String product = TextProduct.Text;
+            String time = TextTime.Text;
+            var csv = new StringBuilder();
+
+
+            // Setting person's ID
+            personsID = personsID + personID.Count;
+
+            var newLine = string.Format("{0},{1},{2},{3}", product, time, MainMap.TargetCenter, personsID);
+
+            csv.AppendLine(newLine);
+
+            File.AppendAllText(filePath, csv.ToString());
+        }
+
+        private void ButtonCancel_Clicked(object sender, RoutedEventArgs e)
+        {
+            GridSellerAdd.Visibility = Visibility.Collapsed;
+            SYLMap.RemoveLastMarker();
+            if (updateMarketplacesWasClicked)
+                SetMarkers(sender, e);
+        }
+
+        private void SetMarkers(object sender, RoutedEventArgs e)
+        {
+            List<double> listXCoord = new List<double>();
+            List<double> listYCoord = new List<double>();
+            List<int> listpersonsID = new List<int>();
+
+            using (TextFieldParser csvParser = new TextFieldParser(filePath))
+            {
+                csvParser.CommentTokens = new string[] { "#" };
+                csvParser.SetDelimiters(new string[] { "," });
+                csvParser.HasFieldsEnclosedInQuotes = true;
+
+                // Skip the row with the column names
+                csvParser.ReadLine();
+
+                while (!csvParser.EndOfData)
+                {
+                    // Saving data to a list
+                    string[] fields = csvParser.ReadFields();
+                    listXCoord.Add(Double.Parse(fields[2]));
+                    listYCoord.Add(Double.Parse(fields[3]));
+                    listpersonsID.Add(Int32.Parse(fields[4]));
+                }
+            }
```

Comment:
The same goes here. Make a parser somewhere else. In this way later on we can reuse this parser on different platforms- Web or Android